ext/ffi/turbocall.rs

```diff
@@ -910,6 +910,7 @@ impl Aarch64Apple {
     aarch64!(self.assmblr; str x0, [x19]);
   }
 
+  #[allow(clippy::unnecessary_cast)]
```

Comment:
Can't we remove this suppression as well?

Reply:
> Can't we remove this suppression as well?

Yes, I think we remove this suppression is better.
TBH, I have no experience about `aarch64!()`, but I want to resolved it with some help. Could you help me.

Reply:
Removed `#[allow(clippy::unnecessary_cast)]` code has this error info:

```rs
error: casting to the same type is unnecessary (`u32` -> `u32`)
   --> ext\ffi\turbocall.rs:920:28
    |
920 |       ; stp x29, x30, [sp, self.allocated_stack - 16]
    |                            ^^^^ help: try: `dynasm!($assembler; .arch aarch64; $($tokens)+)`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_cast
    = note: `-D clippy::unnecessary-cast` implied by `-D clippy::all`

error: casting to the same type is unnecessary (`u32` -> `u32`)
   --> ext\ffi\turbocall.rs:932:47
    |
932 |     aarch64!(self.assmblr; ldp x29, x30, [sp, self.allocated_stack - 16])
    |                                               ^^^^ help: try: `dynasm!($assembler; .arch aarch64; $($tokens)+)`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_cast

error: could not compile `deno_ffi` due to 2 previous errors
```

Reply:
Seems like a the lint error comes from the macro expansion. I think its fine to keep the clippy suppression here

Reply:
It's really complicated as the proc macro is involved here. Apparently this macro is expanded to something like this:

```rs
{
    assmblr
        .push_u32(
            2835381245u32
                | ((((allocated_stack - 16 >> 3u8) as u32) & 127u32) << 15u8),
        );
};
```

So clippy compains about `as u32` part of this expansion.
I don't think this clippy rule should check for the resulting code generated from macros, so probably we could fix it on the clippy side, but for the time being it would make sense to suppress the lint in my opinion. 
One suggestion though, is that we would like to limit the scope of the suppression. So could you put `#[allow(clippy::unnecessary_cast)]` right before the macro invocation, like:

```rs
#[allow(clippy::unnecessary_cast)]
{
  aarch64!(self.assmblr
    // Frame record is stored at the bottom of the stack frame
    ; stp x29, x30, [sp, self.allocated_stack - 16]
    ; add x29, sp, self.allocated_stack - 16
  )
}
```